### ISSUE TITLE
no automatic synchronization on incoming file changes

### ISSUE THREAD

**Issue by wolfgang371** [39aldo39/DecSyncCC#53]
I have two mobiles, both with DecSync CC and Syncthing. They both use the same DecSync calendar account. Also, they both synchronize on a common server.
Any time I change any item in one of the two calendars it gets synced to my server after some seconds, so this part (automatic sync on outgoing calendar changes) is fine.

However, the calendar of the other mobile doesn't get updated automatically; I have to force it by manually pressing the "refresh" button in DecSync.

**Comment by wolfgang371** [comment]
It seems that there is no _immediate_ synchronization, but sometimes after 30min, sometimes 2h there happens an automatic sync... I probably was not patient enough...

**Comment by cryptono** [comment]
+1, having the same issue.
The main reason right now from keeping me to further use DecSync.

**Comment by wolfgang371** [comment]
Thanks for the update! Meanwhile I use the new version on both mobiles; it still takes sometimes several hours, though.

**Comment by 40417256** [comment]
How about adding a configurable automatic refresh on top of the manual one ?

**Comment by wolfgang371** [comment]
You mean a configurable sync. polling period as a feature?
Hm. From what I recall from my test at the top of this issue it seems there's (currently) no notification back from the syncthing'ed incoming calendar files to DecSyncCC.
On a standard linux there's `inotifywait` - I wonder if it's available on Andoid as well? That would be a more battery efficient solution on the one hand side and a faster one on the other as well.

**Comment by 39aldo39** [comment]
There is not really a good `inotifywait` equivalent on Android. There is something similar (but only for older Android versions I think), but that requires a constant background process which is undesirable.

I looked at the enforced intervals again and apparently the minimum allowed time is now 15 instead of 60 minutes, so I did change that.

**Comment by adam4235** [comment]
But in using Decsync CC I'm also using Syncthing on my phone and computer to sync the files.  That already has a constant background process running to do the sync, right?  And it syncs pretty fast.  So why would be undesirable to add a second background process to sync the files that came from Syncthing with the phones's contacts / calendar / tasks?

**Comment by eode** [comment]
I've put in a request with syncthing-fork to do an intent broadcast, so that this (or other) apps have some way of knowing when a folder has been updated.

**Comment by eode** [comment]
@39aldo39 

I know per #93 you may not have time for this, but in case it interests you:

SyncThing Fork now notifies of folder changes by sending an `Intent`, specifically to help address this issue for DecsyncCC (although any other app can follow the integration process, of course).  Thus, DecsyncCC can listen for the intent, and react when it occurs.

This doesn't fix the issue with non-syncthing updates to the files, but I think Syncthing is a common use case for DecsyncCC.  When Syncthing-Fork notifies, then DecsyncCC could either do a general scan, or a scan of any DecsyncCC folder that is or is within the specified updated folder.

Integration is explained here on [this wiki page](https://github.com/Catfriend1/syncthing-android/blob/main/wiki/Integrate-your-app-with-Syncthing-receive-folder-sync-status.md)

Related Syncthing-Fork issue: Catfriend1/syncthing-android#1410

**Comment by onetimecontributor** [comment]
@eode do you still use Decsync? does it work for you?

**Comment by eode** [comment]
Yes, I do, and yes, it does.  I want some better integration with desktop calendars, but overall, it works well enough.  partner and I sync calendars this way.

if you use Evolution, it's great.
